### ISSUE TITLE
InvalidOperationException in MapView constructor with 5.0.0-beta.4

### ISSUE THREAD

**Issue by janusw** [Mapsui/Mapsui#2783]
**Mapsui Version**
5.0.0-beta.4

**Mapsui Platform**
MAUI

**Device**
Android 14

**Describe the bug**
The MapView class seems to be completely broken in 5.0.0-beta.4. Any attempt to create a MapView results in the following exception (did not happen with beta.3):

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Sequence contains no matching element
  at at System.Linq.ThrowHelper.ThrowNoMatchException()
  at at System.Linq.Enumerable.First[LayerEntry](IEnumerable`1 source, Func`2 predicate)
  at Mapsui.Layers.LayerCollection.RemoveInternal(ILayer[] layers) in Mapsui/Mapsui/Layers/LayerCollection.cs:378
  at Mapsui.Layers.LayerCollection.Remove(ILayer[] layers) in Mapsui/Mapsui/Layers/LayerCollection.cs:229
  at Mapsui.UI.Maui.MapView.RemoveLayers() in Mapsui/Mapsui.UI.MapView/MapView.cs:712
  at Mapsui.UI.Maui.MapView.OnPropertyChanged(String propertyName) in Mapsui/Mapsui.UI.MapView/MapView.cs:411
  at Microsoft.Maui.Controls.BindableObject.SetValueActual(BindableProperty property, BindablePropertyContext context, Object value, Boolean currentlyApplying, SetValueFlags attributes, SetterSpecificity specificity, Boolean silent) in /_/src/Controls/src/Core/BindableObject.cs:642
  at Microsoft.Maui.Controls.BindableObject.SetValueCore(BindableProperty property, Object value, SetValueFlags attributes, SetValuePrivateFlags privateAttributes, SetterSpecificity specificity) in /_/src/Controls/src/Core/BindableObject.cs:569
  at Microsoft.Maui.Controls.BindableObject.SetValue(BindableProperty property, Object value) in /_/src/Controls/src/Core/BindableObject.cs:474
  at Mapsui.UI.Maui.MapControl.get_Map() in Mapsui/Mapsui.UI.Shared/MapControl.cs:427
  at Mapsui.UI.Maui.MapView.OnPropertyChanged(String propertyName) in Mapsui/Mapsui.UI.MapView/MapView.cs:363
  at Microsoft.Maui.Controls.BindableObject.SetValueActual(BindableProperty property, BindablePropertyContext context, Object value, Boolean currentlyApplying, SetValueFlags attributes, SetterSpecificity specificity, Boolean silent) in /_/src/Controls/src/Core/BindableObject.cs:642
  at Microsoft.Maui.Controls.BindableObject.SetValueCore(BindableProperty property, Object value, SetValueFlags attributes, SetValuePrivateFlags privateAttributes, SetterSpecificity specificity) in /_/src/Controls/src/Core/BindableObject.cs:569
  at Microsoft.Maui.Controls.BindableObject.SetValue(BindablePropertyKey propertyKey, Object value) in /_/src/Controls/src/Core/BindableObject.cs:489
  at Microsoft.Maui.Controls.NavigableElement.set_Navigation(INavigation value) in /_/src/Controls/src/Core/Shell/NavigableElement.cs:37
  at Microsoft.Maui.Controls.NavigableElement..ctor() in /_/src/Controls/src/Core/Shell/NavigableElement.cs:28
  at Microsoft.Maui.Controls.VisualElement..ctor() in /_/src/Controls/src/Core/VisualElement/VisualElement.cs:517
  at Microsoft.Maui.Controls.View..ctor() in /_/src/Controls/src/Core/View/View.cs:94
  at Microsoft.Maui.Controls.Compatibility.Layout..ctor() in /_/src/Controls/src/Core/LegacyLayouts/Layout.cs:109
  at at Microsoft.Maui.Controls.TemplatedView..ctor()
  at at Microsoft.Maui.Controls.ContentView..ctor()
  at Mapsui.UI.Maui.MapControl..ctor() in Mapsui/Mapsui.UI.Maui/MapControl.cs:41
  at Mapsui.UI.Maui.MapView..ctor() in Mapsui/Mapsui.UI.MapView/MapView.cs:43
```

**To Reproduce**
1. Build and run the `Mapsui.Samples.Maui.MapView` sample app.
2. Select any category (e.g. "Demo).
3. Select any sample (e.g. "OpenStreetMap" or any other one).
4. The app crashes with an InvalidOperationException.

**Expected behavior**
No exception.

**Comment by janusw** [comment]
Most probably related to PR #2762 by @pauldendulk.

**Comment by janusw** [comment]
> Most probably related to PR #2762 by @pauldendulk.

in particular commit 558c16fbfbf4a387f139f2e14288989d777f43ec seems to be the culprit

**Comment by janusw** [comment]
Gladly, it seems like this issue is fixed on the main branch already by f5dfc46ac5b46a751de46427c1a109047cfe5ba9. Closing.

**Comment by pauldendulk** [comment]
Before my LayerCollection changes removing a non-existing layer did not crash but returned a false boolean. This is not a great interface because it returns a single bool for removing a layer collection, which could have one or multiple non existing layers, but breaking it was just a mistake I made. So, I fixed that in this PR https://github.com/Mapsui/Mapsui/pull/2784
